processor/transform/compatability.ts

```diff
@@ -19,6 +20,12 @@ const compatibilityTransfomer = (): Transform => tree => {
     return SKIP;
   });
 
+  visit(tree, 'image', (node: Image, index: number, parent: Parent) => {
+    if (phrasing(parent) || !parent.children.every(child => child.type === 'image' || !phrasing(child))) return;
```

Comment:
Wait, would it match this:

```md
Lorem ![](/my.img) ipsum.
```

And does that mean we'd be treating/displaying ^said image as a block? That doesn't seem right…

Reply:
And also, would this image still float to the right?

```md
Lorem ipsum:

<img align=right src=my.img />

Dolor set amos.
```

Reply:
> And does that mean we'd be treating/displaying ^said image as a block? That doesn't seem right…

We've been treating images that are alone on a line as a readme-style block level image for sometime:
```
## Block Image

![](my.img)

Lorem ipsum.
```

> And also, would this image still float to the right?

Yea, that would be parsed as `mdxJsxFlowElement`. The `image` node type is _usually_ only used for markdown images. RDMD would convert magic blocks to `image` nodes for simplicity. But for the reasons you're asking about, was probably a huge mistake.

Reply:
I guess, should we use this migration to convert lone markdown images to JSX, to remove the ambiguity?

Reply:
> And does that mean we'd be treating/displaying ^said image as a block? That doesn't seem right…

THis logic should ignore that image then. It's checking to make sure that all of the images siblings aren't phrasing. eg:

```
![](my.img)

## Heading

![](my-2.img)

> ![](my-3.img)
>
> - all block components
```
```